### PR TITLE
Fix milestoning date propagation bug

### PR DESCRIPTION
**Pull request by gayathrir11**
#### What type of PR is this?

Bug fix

#### What does this PR do / why is it needed ?

This PR fixes bug with milestoning date propagation. 

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?
Yes

### CODE REVIEW

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperValueSpecificationBuilder.java [hunk 1]
```diff
@@ -207,8 +224,8 @@ else if (foundProperty instanceof QualifiedProperty)
                 List<Variable> lambdaParams = new FastList<>();
                 lambdaParams.add(automaLambdaparam);
                 automapLambda.parameters = lambdaParams;
-                MilestoningDatePropagationHelper.updateMilestoningPropagationContextForAutoMap(foundProperty, context, appliedProperty.parameters.size(), processedParameters.get(0), processingContext);
```
Comment (rafaelbey) [inline legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperValueSpecificationBuilder.java:210]: Is this method use anywhere else?  I did not find any usage, hence if that is correct, lets clean up the code of unused methods.

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/MilestoningDatePropagationHelper.java [hunk 2]
```diff
@@ -68,24 +69,38 @@ public static void setMilestoningDates(ListIterable<? extends ValueSpecification
     {
         if (isBusinessTemporal(milestoningStereotype))
         {
-            processingContext.milestoningDatePropagationContext.setBusinessDate(temporalParameterValues.get(1));
+            ValueSpecification businessdate = null;
+            if (temporalParameterValues.size() > 1)
```
Comment (rafaelbey) [inline legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/MilestoningDatePropagationHelper.java:73]: Checking if parameters are **_greater than_** seems odd to me.  In what case defining more than expected is correct?  

Even further, if the modeled class does not have any temporal stereotype, should we validate for zero parameters?

I recall seeing users defining temporal argument on non-temporal classes without any failure.

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/ProcessingContext.java [hunk 3]
```diff
@@ -26,8 +26,25 @@
...
 {
     public MutableList<MutableMap<String, ValueSpecification>> inferredVariableList = FastList.newList();
-    public MilestoningDatePropagationContext milestoningDatePropagationContext = new MilestoningDatePropagationContext();
+    public Stack<MilestoningDatePropagationContext> milestoningDatePropagationContext = new Stack<>();
     private final Stack<String> tags = new Stack<>();
+    public boolean isDatePropagationSupported = true;
+
+    public void pushMilestoningDatePropagationContext(MilestoningDatePropagationContext milestoningContext)
+    {
+        MilestoningDatePropagationContext context = new MilestoningDatePropagationContext(milestoningContext.getProcessingDate(), milestoningContext.getBusinessDate());
```
Comment (rafaelbey) [inline legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/ProcessingContext.java:35]: Why creating a new context?  Cannot we make MilestoningDatePropagationContext immutable?  It seems that you remove the need of the setters on with this change.  

Note that you are anyways exposing the stack references thru the peek, hence ensuring state cannot change might be required.

File: legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperValueSpecificationBuilder.java (hunk 1 above)
Comment (gayathrir11) [reply legend-engine-language-pure-compiler/src/main/java/org/finos/legend/engine/language/pure/compiler/toPureGraph/HelperValueSpecificationBuilder.java:210]: done

**Review by rafaelbey — APPROVED** [review]
Thanks for taking care of the feedback items.

**Pull request merged**